static/scripts/pay.ts

```diff
@@ -30,11 +39,22 @@ const withdraw = async (signer: JsonRpcSigner, txData: TxType, predefined: strin
 };
 
 const fetchTreasury = async (): Promise<{ balance: number; allowance: number }> => {
-  const tokenAddress = txData.permit.permitted.token;
-  const tokenContract = new ethers.Contract(tokenAddress, daiAbi, new ethers.providers.Web3Provider((window as any).ethereum));
-  const balance = await tokenContract.balanceOf(txData.owner);
-  const allowance = await tokenContract.allowance(txData.owner, permit2Address);
-  return { balance, allowance };
+  try {
+    const provider = new ethers.providers.Web3Provider((window as any).ethereum)
+
+    const tokenAddress = txData.permit.permitted.token;
+    const tokenContract = new ethers.Contract(tokenAddress, daiAbi, provider);
+    const balance = await tokenContract.balanceOf(txData.owner);
+    const allowance = await tokenContract.allowance(txData.owner, permit2Address);
+    return { balance, allowance };
+  } catch (error: any) {
+    if(error?.message?.includes("missing provider")) {
+      alert("Error: Please use a web3 enabled browser.");
+    } else {
+      alert("Error: Please connect your wallet.");
+    }
+    return { balance: 0, allowance: 0 };
```

Comment:
I see that you want to render `0` for both. Unfortunately this is made up value and is not accurate information. Instead I think it makes sense to hide the table row entirely. Or for now, it could be simpler to write "N/A" 

If types are too tedious, even a `-1` would signal an error more clearly as negative for either value is impossible.